app/Metadata/Extractor.php

```diff
@@ -10,6 +10,15 @@
 
 class Extractor
 {
+	/*
+	 * https://github.com/LycheeOrg/Lychee/issues/1200
+	 *
+	 * Minimum possible recordable date value.
+	 *
+	 * Equates to 0000:01:01 12:00:00
+	 */
+	private const MINIMUM_VALID_DATE_UTC = -62167176000;
```

Comment:
Why is this the minimum valid UTC date? What is about dates BC? Sure, there were no digital cameras but please consider digitized drawings with artificially created EXIF data which points to the time when the original drawing has presumably be created. This is what some national archives do.

Please provide a rational for this number.

Also note, that this number cannot be represented by 32bit and we still have users with 32bit installations. This code won't work for them.

Reply:
Couple of things:

1. The minimum possible valid date the `CreateDate`, `ModifyDate`, and `DateTimeOriginal` exif fields can go is `0000:01:01 00:00:00`.  They are not intended to handle BC dates.
2. If we were worried about being able to handle BC dates, that should be a different unit of work that looks at the XMP fields defined here: https://exiftool.org/TagNames/XMP.html#iptcExt
    * http://www.iptc.org/std/photometadata/specification/IPTC-PhotoMetadata#circa-date-created for some more reading
3. Good point, it should be a string:
    ```
    $ uname -m
    armv7l
    $ php -r 'var_dump(PHP_INT_MAX, date_create_from_format("U", "-62167176000"));'
    int(2147483647)
    object(DateTime)#1 (3) {
      ["date"]=>
      string(26) "0000-01-01 12:00:00.000000"
      ["timezone_type"]=>
      int(1)
      ["timezone"]=>
      string(6) "+00:00"
    }
    ```
4. Zero and negative year values (indicating a Gregorian BC year) is not supported by Postgres when inserting into a timestamp field using an iso 8601 string value without first wrapping the value in a `to_timestamp` function call, as such:
    ```
    # select to_timestamp('-0001-11-30', 'YYYY');
              to_timestamp        
     ---------------------------
     0001-01-01 00:00:00+14 BC
    ```
    * This is from a test container where I set the TZ to `UTC+14` to see how it behaved
    * As an example, this works:
    ```
    # insert into photos (id, legacy_id, title, type, checksum, original_checksum, created_at, updated_at, taken_at) values (1, 1, 'test1', 'photo', 'asdf', 'asdf', now(), now(), to_timestamp('-0001-01-01', 'YYYY-MM-DD bc'));
    INSERT 0 1
    ```
    * But this doesn't:
    ```
    # insert into photos (id, legacy_id, title, type, checksum, original_checksum, created_at, updated_at, taken_at) values (1, 1, 'test1', 'photo', 'asdf', 'asdf', now(), now(), '-0001-01-01');
    ERROR:  invalid input syntax for type timestamp: "-0001-01-01"
    LINE 1: ..., 'test1', 'photo', 'asdf', 'asdf', now(), now(), '-0001-01-..
    ```
    * There has been some [recent discussion](https://www.postgresql.org/message-id/9q0367bkf9b.fsf%40gmx.us) on supporting `+/-` prefix on on number values in pgsql iso 8601 parsing, so maybe this will change in the future (but I doubt it).
5. I chose that value after some experimentation 
6. Zero year values _are_ supported by MariaDB's timestamp columns, but negative values are not.

I ultimately chose that value after some local experimentation with sqlite and a MariaDB container, however after the [initial test failed](https://github.com/LycheeOrg/Lychee/runs/4844977600?check_suite_focus=true#step:17:75) I tried to quickly update before I had to head out to run some errands.

I've since spent some additional time testing locally with pgsql, and it seems the minimum possible iso 8601 date value pgsql will accept is `0001:01:01 00:00:00`.

Reply:
Your argument suffers from the implicit assumption that one has to use ISO 8601 strings.

Only because a specific DB (e.g. PostgreSQL) does not support to insert a date like "-0009-14-01 17:34:00" with a negative date part does not mean that these dates aren't supported at all. I am even not sure if this would count for a valid ISO 8601 steing. The lowest by PostgreSQL supported `DateTime` is something around 4000 BC. You yourself gave an example how to insert a "negative" date into PostgreSQL: you have to use the suffix BC instead of a prefix sign operator.

Reply:
Couple of things:

1. As I mentioned, MariaDB also cannot accept negative ISO 8601 values.

2. I would be all for modifying the serialization of timestamp columns to be something a bit more flexible, however there are a few things to consider here:
    1. As I mentioned above, The exif date fields _will never be negative_
    1. I think it may make sense to increase the # of date fields represented in the DB so Lychee can handle those extended exif date fields I mentioned above.
    2. This issue is _purely_ to prevent improper interpolation of exif date values